### PR TITLE
chore: remove duplicate code from dummy-module

### PR DESCRIPTION
**Pull request by ngutech21**
Cleans up duplicated code in the dummy-modules:
* Remove ServerModuleGen and ServerModule from dummy-common
* Remove DummyVerificationCache from dummy-server and use the DummyVerificationCache from the dummy-common module instead

### CODE REVIEW

File: modules/fedimint-dummy-server/src/lib.rs [hunk 1]
```diff
@@ -247,8 +247,3 @@ impl Dummy {
         Dummy { cfg }
     }
 }
-
-#[derive(Debug, Clone)]
-pub struct DummyVerificationCache;
```
Comment (dpc) [inline modules/fedimint-dummy-server/src/lib.rs:252]: Verification cache should be a part of a server. Client has no need for it. So I would delete the duplicate in common.

**Review by dpc — COMMENTED** [review]
No idea how this happen. Thanks for taking care of it. Other than verification cache thing, LGTM

File: modules/fedimint-dummy-server/src/lib.rs (hunk 1 above)
Comment (ngutech21) [reply modules/fedimint-dummy-server/src/lib.rs:252]: Yes makes sense

File: modules/fedimint-dummy-common/src/lib.rs [hunk 2]
```diff
@@ -94,12 +91,11 @@ impl ModuleCommon for DummyModuleTypes {
     type ConsensusItem = DummyConsensusItem;
 }
 
-plugin_types_trait_impl!(
```
Comment (dpc) [inline modules/fedimint-dummy-common/src/lib.rs:335]: We need to delete `plugin_types_trait_impl` BTW. Only `plugin_types_trait_impl_common` should be needed now.

**Review by dpc — DISMISSED** [review]

**Review by elsirion — DISMISSED** [review]
That's the best kind of PR: removing instead of adding code ^^

**Comment by elsirion** [comment]
Oh, needs rebase :(

**Review by dpc — APPROVED** [review]

**Review by elsirion — APPROVED** [review]

**Comment by justinmoon** [comment]
Nice PR!

**Pull request merged**